packages/framework/src/Model/Transport/Transport.php

```diff
@@ -242,4 +251,52 @@ protected function createTranslation()
     {
         return new TransportTranslation();
     }
+
+    /**
+     * @param \Shopsys\FrameworkBundle\Model\Payment\Payment $payment
+     */
+    public function addPayment(Payment $payment)
+    {
+        if (!$this->payments->contains($payment)) {
+            $this->payments->add($payment);
+            $payment->addTransport($this);
+        }
+    }
+
+    /**
+     * @param \Shopsys\FrameworkBundle\Model\Payment\Payment[] $payments
+     */
+    public function setPayments(array $payments)
+    {
+        $this->clearPayments();
+
+        foreach ($payments as $payment) {
+            $this->addPayment($payment);
+        }
+    }
+
+    /**
+     * @param \Shopsys\FrameworkBundle\Model\Payment\Payment $payment
+     */
+    public function removePayment(Payment $payment)
+    {
+        $this->payments->removeElement($payment);
+        $payment->removeTransport($this);
+    }
+
+    protected function clearPayments()
+    {
+        foreach ($this->payments as $payment) {
+            $this->removePayment($payment);
+            $payment->removeTransport($this);
```

Comment:
`removePayment` calls also `$payment->removeTransport($this)` This duplication will fail